src/hotspot/share/gc/shared/space.cpp

```diff
@@ -51,20 +51,18 @@ HeapWord* DirtyCardToOopClosure::get_actual_top(HeapWord* top,
                                                 HeapWord* top_obj) {
   if (top_obj != NULL) {
     if (_sp->block_is_obj(top_obj)) {
-      if (_precision == CardTable::ObjHeadPreciseArray) {
-        if (cast_to_oop(top_obj)->is_objArray() || cast_to_oop(top_obj)->is_typeArray()) {
-          // An arrayOop is starting on the dirty card - since we do exact
-          // store checks for objArrays we are done.
-        } else {
-          // Otherwise, it is possible that the object starting on the dirty
-          // card spans the entire card, and that the store happened on a
-          // later card.  Figure out where the object ends.
-          // Use the block_size() method of the space over which
-          // the iteration is being done.  That space (e.g. CMS) may have
-          // specific requirements on object sizes which will
-          // be reflected in the block_size() method.
-          top = top_obj + cast_to_oop(top_obj)->size();
-        }
+      if (cast_to_oop(top_obj)->is_objArray() || cast_to_oop(top_obj)->is_typeArray()) {
+        // An arrayOop is starting on the dirty card - since we do exact
+        // store checks for objArrays we are done.
+      } else {
+        // Otherwise, it is possible that the object starting on the dirty
+        // card spans the entire card, and that the store happened on a
+        // later card.  Figure out where the object ends.
+        // Use the block_size() method of the space over which
+        // the iteration is being done.  That space (e.g. CMS) may have
```

Comment:
What is "CMS" here?

Reply:
The code uses `size()` rather than `block_size()`. So my guess is this comment is somewhat obsolete and the comment verbage would need minor updates.

Reply:
Indeed, I'd delete lines 61-64 and be done.

Reply:
I think @ysramakrishna is right, and agree with his suggestion to delete stale comment lines.